src/OpenApi/gen/XmlComments/MemberKey.cs

```diff
@@ -32,16 +32,10 @@ public static MemberKey FromMethodSymbol(IMethodSymbol method, Compilation compi
         {
             // Handle Task/ValueTask for async methods
             var actualReturnType = method.ReturnType;
-            if (method.IsAsync && actualReturnType is INamedTypeSymbol namedType)
+            if (method.IsAsync
+                && actualReturnType is INamedTypeSymbol { TypeArguments.Length: 1 } namedType)
             {
-                if (namedType.TypeArguments.Length > 0)
-                {
-                    actualReturnType = namedType.TypeArguments[0];
-                }
-                else
-                {
-                    actualReturnType = compilation.GetSpecialType(SpecialType.System_Void);
-                }
+                actualReturnType = namedType.ConstructedFrom;
```

Comment:
Was trying to understand what exactly this did, and why we couldn't just get rid of this check since `ReplaceGenericArguments` below should handle `Task<T>`.
Looks like `ConstructedFrom` would return `Task<int>` if you had a type like `Example : Task<int>`. Is that the desired effect? We should probably add a test like that.

We also have tests for `Task` and `Task<T>`
https://github.com/dotnet/aspnetcore/blob/394df7ad2d62639e5434e4b7def133a34ccb5ed1/src/OpenApi/test/Microsoft.AspNetCore.OpenApi.SourceGenerators.Tests/snapshots/CompletenessTests.SupportsAllXmlTagsOnSchemas%23OpenApiXmlCommentSupport.generated.verified.cs#L274-L276
So not too sure what was broken.